userspace/libsinsp/event.cpp

```diff
@@ -188,16 +188,14 @@ uint32_t sinsp_evt::get_num_params()
 	return (uint32_t)m_params.size();
 }
 
-sinsp_evt_param *sinsp_evt::get_param(uint32_t id)
+sinsp_evt_param *sinsp_evt::get_param_buffer(uint32_t id)
 {
 	if((m_flags & sinsp_evt::SINSP_EF_PARAMS_LOADED) == 0)
 	{
 		load_params();
 		m_flags |= (uint32_t)sinsp_evt::SINSP_EF_PARAMS_LOADED;
 	}
 
-	ASSERT(id < m_params.size());
-
 	return &(m_params.at(id));
```

Comment:
Can't this method return directly the buffer, ie: `->m_val`?
Given its name, i'd expect that.
And when we need to get the full `sinsp_evt_param` we should call a new method IMHO (or a `get_param` without template arg perhaps?)

Reply:
> Can't this method return directly the buffer, ie: ->m_val?
> Given its name, i'd expect that.

A buffer has a pointer and a size, and that's what the functions that call this are expecting. TBH, instead of a `sinsp_evt_param` I would prefer a `scap_const_sized_buffer` but if I did that change this PR would be even bigger :)

I could collapse this into `get_param()` (with no template) but does C++ allow me to do overload a template with a non template? I honestly don't know.

However, you might want a `const char*` string (NUL-terminated sequence of characters), that's why I added the `get_param_const_char` method, which only returns a pointer to the string and also checks the bounds of the string against the length to make sure it is not reading garbage.

Reply:
> I could collapse this into get_param() (with no template) but does C++ allow me to do overload a template with a non template? I honestly don't know.

I would be a fan of this! Anyway, i understood your point, thank you!

Reply:
> I would be a fan of this!

It looks like it's possible :) If it builds and passes tests I will commit that. I like it because in order to access parameters you _need_ to call that function because it will load params if it doesn't have them. This makes it more intuitive to use I think.

Reply:
Done! This means that I do not need to change many `get_param()` calls with `get_param_buffer()`. PTAL :)

Reply:
Yay thank you!